ATTACK/Assets/Scripts/CardManager.cs

```diff
@@ -0,0 +1,102 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public struct Card
+{
+    public int position;
+    public int rank;
+    public bool rotated;
+
+    public static Card INVALID = new Card(-1, -1, false);
+
+    public Card(int position, int rank, bool rotated)
+    {
+        this.position = position;
+        this.rank = rank;
+        this.rotated = rotated;
+    }
+
+    public override string ToString()
+    {
+        return $"position: {position}, rank: {rank}, rotated: {rotated}";
+    }
+}
+public class CardManager : MonoBehaviour
+{
+    private static CardManager Instance;
+
+    [HideInInspector]
+    public const int MAX_CARDS_PER_TEAM = 5;
+
+    private Card[] T1Cards;
+    private Card[] T2Cards;
+
+    void Awake()
+    {
+        if (Instance == null)
+            Instance = this;
+        else
+            Destroy(this);
+
+        T1Cards = new Card[MAX_CARDS_PER_TEAM];
+        for (int i = 0; i < T1Cards.Length; i++) 
+            T1Cards[i] = Card.INVALID;
+        T2Cards = new Card[MAX_CARDS_PER_TEAM];
+        for (int i = 0; i < T2Cards.Length; i++) 
+            T2Cards[i] = Card.INVALID;
+    }
+
+    void Update()
+    {
+        UpdateCards();
+    }
+
+    public static bool HasCard(Team team, int index)
+    {
+        if (index < 0 || index >= MAX_CARDS_PER_TEAM)
+            return false;
+        else if (team == 0)
+            return !Equals(Instance.T1Cards[index], Card.INVALID);
+        else
+            return !Equals(Instance.T2Cards[index], Card.INVALID);
+    }
+
+    public static Card GetCard(Team team, int index)
+    {
+        if (team == Team.One)
+            return Instance.T1Cards[index];
+        return Instance.T2Cards[index];
+    }
+
+    private void UpdateCards()
+    {
+        for (int i = 0; i < T1Cards.Length; i++)
```

Comment:
Nicely spotted. I probably rushed through this. Currently this code is broken, since we need to update both teams' cards, and right now we only update T1. (Can also be seen in the way we only set T1Cards[i] to Invalid if it was not found).  I'll fix this properly.